pkg/extensions/sync/signatures.go

```diff
@@ -372,34 +372,34 @@ func (sig *signaturesCopier) syncOCIRefs(localRepo, remoteRepo, digestStr string
 		}
 
 		if sig.repoDB != nil {
-			sig.log.Debug().Str("repository", localRepo).Str("digest", digestStr).Msg("trying to add OCI refs for repo digest")
+			sig.log.Debug().Str("repository", localRepo).Str("digest", subjectStr).Msg("trying to add OCI refs for repo digest")
 
-			isSig, _, signedManifestDig, err := storage.CheckIsImageSignature(localRepo, OCIRefBody, ref.Digest.String())
+			isSig, sigType, signedManifestDig, err := storage.CheckIsImageSignature(localRepo, OCIRefBody, ref.Digest.String())
 			if err != nil {
-				return fmt.Errorf("failed to set metadata for OCI ref in '%s@%s': %w", localRepo, digestStr, err)
+				return fmt.Errorf("failed to set metadata for OCI ref in '%s@%s': %w", localRepo, subjectStr, err)
 			}
 
 			if isSig {
 				err = sig.repoDB.AddManifestSignature(localRepo, signedManifestDig, repodb.SignatureMetadata{
-					SignatureType:   repodb.NotationType,
-					SignatureDigest: digestStr,
+					SignatureType:   sigType,
+					SignatureDigest: refDigest.String(),
 				})
 			} else {
-				err = repodb.SetImageMetaFromInput(localRepo, digestStr, ref.MediaType,
-					digest, OCIRefBody, sig.storeController.GetImageStore(localRepo),
+				err = repodb.SetImageMetaFromInput(localRepo, subjectStr, ref.MediaType,
```

Comment:
instead of subjectStr, shouldn't be refDigest ?

Reply:
the digest of the oci reference itself, not parent.

Reply:
Yes this is correct. I've updated it